src/main/java/hudson/plugins/git/extensions/impl/SubmoduleOption.java

```diff
@@ -45,15 +45,17 @@
     /** Use --depth flag on submodule update command - requires git>=1.8.4 */
     private boolean shallow;
     private Integer depth;
+    private Integer threads = 1;
```

Comment:
I wouldn't use a default value here. Due to `setThreads(Integer threads)` accepting `null` and negative numbers, a proper value check is anyway required before handing it to `SubmoduleUpdateCommand::threads`.

Please remove the default value assignment.